feat: Added an option to enable/disable chat models.

anpigon opened the pull request:
## Description

A new feature has been added to allow users to turn chat models on or off as needed.

Users can toggle the enable button in the settings to switch the chat model between enabled and disabled states.

<img width="778" alt="스크린샷 2025-02-09 오후 12 22 10" src="https://github.com/user-attachments/assets/68822d69-e9c9-4914-9c3b-0f4fa75daf52" />


## Checklist before requesting a review
- [x] I have reviewed the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [x] I have performed a self-review of my code
- [x] I have performed a code linting check and type check (by running `npm run lint:check` and `npm run type:check`)

glowingjade commented on src/components/settings/sections/models/ChatModelsSubSection.tsx:
```diff
@@ -52,6 +72,19 @@ export function ChatModelsSubSection({
                 <td>{chatModel.providerId}</td>
                 <td>{chatModel.model}</td>
+                <td>
+                  <div
+                    className={clsx('checkbox-container', {
+                      'is-enabled': isEnabled(chatModel.enable),
+                    })}
+                    onClick={() => handleToggleEnableChatModel(chatModel.id)}
+                  >
+                    <input
+                      type="checkbox"
+                      checked={isEnabled(chatModel.enable)}
+                    />
+                  </div>
```
You can use https://github.com/glowingjade/obsidian-smart-composer/blob/main/src/components/common/ObsidianToggle.tsx component for this

anpigon replied:
> You can use https://github.com/glowingjade/obsidian-smart-composer/blob/main/src/components/common/ObsidianToggle.tsx component for this

Thank you for the suggestion! I've updated the code to use the ObsidianToggle component and simplified the logic by removing the negation.

glowingjade commented on src/components/settings/sections/models/ChatModelsSubSection.tsx:
```diff
@@ -31,18 +34,35 @@ export function ChatModelsSubSection({
     })
   }
 
+  const handleToggleEnableChatModel = async (modelId: string) => {
+    await setSettings({
+      ...settings,
+      chatModels: [...settings.chatModels].map((v) =>
+        v.id === modelId ? { ...v, enable: !isEnabled(v.enable) } : v,
+      ),
+    })
+  }
```
If you use ObsidianToggle component, there’s no need to define it using negation. Simply using the value makes it clearer.

glowingjade replied:
Additionally, the function should handle the case where the user attempts to disable a model that is currently selected as the Chat Model or Apply Model. It should display a notice and do nothing, similar to `handleDeleteChatModel`.

anpigon replied:
> Additionally, the function should handle the case where the user attempts to disable a model that is currently selected as the Chat Model or Apply Model. It should display a notice and do nothing, similar to `handleDeleteChatModel`.

Thank you for catching this edge case! I've added a validation check to prevent disabling a model that is currently selected as either the Chat Model or Apply Model. The function now shows a notice to the user in such cases, following the same pattern as handleDeleteChatModel.

glowingjade reviewed (changes requested):
Thanks for the contribution! I’ve added some comments—please take a look and let me know your thoughts.
And the PR title and description should be written in English. I’ll update them this time.

anpigon commented:
> Thanks for the contribution! I’ve added some comments—please take a look and let me know your thoughts.
> And the PR title and description should be written in English. I’ll update them this time.

Thank you for reviewing the PR and for your  feedback! I appreciate the time you took to share your insights.

Apologies for the oversight with the title and description not being in English initially—thank you for updating them! I’ll keep this in mind for future contributions.

Once I’ve incorporated the changes, I’ll let you know.

glowingjade commented:
Thanks for your contribution! Merging this PR. 👍
